### PR TITLE
BUG: Input Metric's original constraint is not being propogated

### PR DESCRIPTION
**Pull request by WilliamDee**
## Context
In derived metric, there are multiple layers of constraints.
1. The constraint in the metric used as an input metric
2. The input_metric constraint defined in the derived metric definition

`(1)` is not being respected so if one were to define a derived metric as so,
```yaml
metric:
  name: "instant_bookings"
  description: "bookings metric"
  type: measure_proxy
  type_params:
    measures:
      - bookings
  constraint: is_instant
---
metric:
  name: "double_instant_bookings"
  description: "2x bookings metric"
  type: derived
  type_params:
    expr: instant_bookings * 2
    metrics:
      - name: instant_bookings
```
The `constraint: is_instant` clause would be missing in the metric `double_instant_bookings`. This PR fixes this bug.

### CODE REVIEW

File: metricflow/test/integration/test_cases/itest_metrics.yaml [hunk 1]
```diff
@@ -803,4 +803,70 @@ integration_test:
...
+      INNER JOIN (
+        SELECT
+          SUM(booking_value) AS booking_value
+          , ds AS metric_time
+        FROM {{ source_schema }}.fct_bookings
+        GROUP BY
+          ds
+      ) b
+      ON a.metric_time = b.metric_time
+    )
```
Comment (tlento) [inline metricflow/test/integration/test_cases/itest_metrics.yaml:872]: This needs an alias

**Review by tlento — COMMENTED** [review]
Seems reasonable! Test needs a fix first.

Would you mind squashing this into a single commit either here or on merge? I don't think we need to add the tests separately (and it'll get rid of the empty commit you added for clabot).

**Review by tlento — COMMENTED** [review]
One more thought, sorry, off my game. I wish there was a better way to make it more obvious why we need both of these things and why they're different, but I can't come up with it right now so comments will have to do.

File: metricflow/model/semantics/metric_semantics.py [hunk 2]
```diff
@@ -128,17 +128,33 @@ def metric_input_specs_for_metric(self, metric_reference: MetricReference) -> Se
         input_metric_specs: List[MetricSpec] = []
 
         for input_metric in metric.input_metrics:
-            spec_constraint = (
+            input_metric_obj = self.get_metric(input_metric.as_reference)
```
Comment (tlento) [inline metricflow/model/semantics/metric_semantics.py:131]: nit: maybe we call this `original_metric_obj` to match the `original_metric_constraint` naming below? I guess I should've noticed this the first time....

**Review by tlento — APPROVED** [review]

File: metricflow/model/semantics/metric_semantics.py (hunk 2 above)
Comment (WilliamDee) [reply metricflow/model/semantics/metric_semantics.py]: done!
